types.go

```diff
@@ -15,33 +15,40 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by the optimizer to specify information needed
-// from the objective function.
-type EvaluationType int
+// EvaluationType is used by the optimizer to specify information needed from
```

Comment:
EvaluationType is used by a Method to specify the objective-function information needed at an x location. The types...